hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4565,3 +4592,24 @@ func generateKubevirtInfraClusterClient(ctx context.Context, cpClient client.Cli
 
 	return infraClusterClient, nil
 }
+
+func (r *HostedClusterReconciler) discoverKubevirtClusterClient(ctx context.Context, hc *hyperv1.HostedCluster) (*kubevirtInfraCluster, error) {
+	loaded, ok := r.kubevirtInfraClusters.Load(hc.Spec.InfraID)
+	if ok {
+		return loaded.(*kubevirtInfraCluster), nil
+	}
+	cluster := &kubevirtInfraCluster{}
+	if hc.Spec.Platform.Kubevirt.Credentials == nil {
+		cluster.Client = r.Client
+		cluster.namespace = hc.Namespace
+	} else {
+		targetClient, err := generateKubevirtInfraClusterClient(ctx, r.Client, *hc)
+		if err != nil {
+			return nil, err
+		}
+		cluster.Client = targetClient
+		cluster.namespace = hc.Spec.Platform.Kubevirt.Credentials.InfraNamespace
+	}
+	r.kubevirtInfraClusters.LoadOrStore(hc.Spec.InfraID, cluster)
```

Comment:
make sure to remove the item from the map on HC deletion. look for the `if !hcluster.DeletionTimestamp.IsZero() ` line, just make sure the map is cleaned up before the finalizer is removed on the HC